### PR TITLE
Multiple notification format

### PR DESCRIPTION
**Pull request by fabgui**
add notification formats, incl time and price
add notification options, incl minimum TGTG quantity
add html formating to Telegram notifications + add /help command for bot.

### CODE REVIEW

**Comment by marklagendijk** [comment]
@fabgui first of all, thanks for your pull request.
I'm trying to keep the code of this project of high quality (as most open source projects will do). As such I cannot merge this pull request like this. 
I'm only replying now, because I don't have much time for this project, and replying to something like this takes quite a bit of time.

You approach to this feature is quite messy. To do this properly you would need to pass the `businesses` to `telegramBot.notify` and handle the HTML there (instead of stripping it out for the other cases).

Also this pull requests includes the changes from your previous pull request. To prevent this you need to create a separate branch whenever you want to create a new pull request. This way the author of the project can choose to merge one pull request, but not the other.

I'll add some comments to the code.

File: config.defaults.json [hunk 1]
```diff
@@ -12,9 +12,12 @@
         "authenticationIntervalInMS": 3600000
     },
     "notifications": {
+        "format": 1,
+        "min_stock": 1,
```
Comment (marklagendijk) [inline config.defaults.json:16]: Unrelated change.

File: config.defaults.json [hunk 2]
```diff
@@ -12,9 +12,12 @@
...
         "console": {
             "enabled": true,
-            "clear": true
+            "clear": true,
+            "include_time": true
```
Comment (marklagendijk) [inline config.defaults.json:20]: Unrelated change.

File: lib/notifier.js [hunk 3]
```diff
@@ -2,20 +2,22 @@ const notifier = require('node-notifier');
...
 };
 
 function notifyIfChanged(businesses) {
-    const message = createMessage(businesses);
     const options = config.get('notifications');
+    var message = createMessage(businesses,options);
+    message = (!message)?'<i>no good TGTG grabs available right now</i>':message;
     if(options.console.enabled){
-        notifyConsole(message, options.console);
+        notifyConsole(message.replace( /(<([^>]+)>)/ig, ''), options.console);
```
Comment (marklagendijk) [inline lib/notifier.js:16]: Stripping out the HTML is of course an ugly approach. There are 2 ways to solve this properly:
1. Pass the businisses to `telegramBot.notify` so it can create its own message.
2. Create a separate `createHtmlMessage` function.

File: lib/notifier.js [hunk 4]
```diff
@@ -25,18 +27,29 @@ function notifyIfChanged(businesses) {
 }
 
 function notifyConsole(message, options){
+    let date_ob = new Date();
     if(options.clear){
         console.clear();
     }
+    if(options.include_time) {
```
Comment (marklagendijk) [inline lib/notifier.js:34]: This feature might make sense, but it would require a separate pull request. Never combine different features in a single pull request, unless you discussed it with the author in advance.

File: lib/notifier.js [hunk 5]
```diff
@@ -25,18 +27,29 @@ function notifyIfChanged(businesses) {
 }
 
 function notifyConsole(message, options){
+    let date_ob = new Date();
```
Comment (marklagendijk) [inline lib/notifier.js:30]: 1. Use `const` when you don't reassign.
2. Use camelCasing for variable names.
3. Don't include type in variable names.
4. Give descriptive names (a new Date gives the current date/time, so the name should reflect that).
```
const now = new Date();
```

File: lib/notifier.js [hunk 6]
```diff
@@ -25,18 +27,29 @@ function notifyIfChanged(businesses) {
...
 }
 
-function createMessage(businesses){
+function createMessage(businesses,options){
     return businesses
-        .map(business => `${ business.display_name } - ${business.items_available}`)
-        .join('\n');
+        .filter(business => (business.items_available >= options.min_stock) )
+        .map(business => (
+			 (options.format==2)? `<b>${ business.display_name }</b> @ ${new Date(Date.parse(business.pickup_interval.start)).toLocaleTimeString('en-GB',business.store.store_time_zone).substring(0,5)}`
```
Comment (marklagendijk) [inline lib/notifier.js:48]: 1. Use a switch statement instead of chaining ternary operators.
2. Use descriptive strings instead of numbers.
```js
switch(options.format){
case 'minimal':
// etc
}
``

File: lib/telegram-bot.js [hunk 7]
```diff
@@ -1,8 +1,12 @@
 const _ = require('lodash');
 const Telegraf = require('telegraf');
+const extra = require('telegraf/extra');
+//const markup = extra.markdown();
+const markup = extra.HTML();
 const { config } = require('./config');
 const cache = {};
 let bot;
+let PausedChats = [];
```
Comment (marklagendijk) [inline lib/telegram-bot.js:9]: 1. This is a separate feature.
2. Variable names should be camelCased instead of TitleCased.

File: lib/telegram-bot.js [hunk 8]
```diff
@@ -13,13 +17,13 @@ function notify(message){
     if(!bot){
         createBot();
     }
-    const chats = config.get('notifications.telegram.chats');
+    const chats = config.get('notifications.telegram.chats').filter(chat => PausedChats.indexOf(chat.id) == -1);
```
Comment (marklagendijk) [inline lib/telegram-bot.js:20]: Use `_.includes` instead of `indexOf`

File: lib/telegram-bot.js [hunk 9]
```diff
@@ -37,35 +41,54 @@ function createBot(){
...
+/help`);
     if(cache.message){
-        context.reply(cache.message);
+        context.reply(cache.message, markup);
     }
 }
 
 function stopCommand(context){
-    context.reply(`*bleep* Ok.. I get it. Too much is too much. I'll stop bothering you now. *bloop*.
+    context.reply(`OK... I'll stop notifying you now.
```
Comment (marklagendijk) [inline lib/telegram-bot.js:63]: Unrelated change.

**Comment by marklagendijk** [comment]
I'll close this pull request. You can open a new one with just the notification formats, and get that one merged after review.

**Comment by fabgui** [comment]
Thanks for this great TGTG watcher and for taking the time to share feedback :)
This was my first attempt at github and first node.js + lodash project; so, learned quite a bit along the way.   I am glad some of it turned useful. Enjoy the end of the summer !
